onnxruntime/contrib_ops/cuda/transformers/generation_cuda_impl.cu

```diff
@@ -759,15 +759,15 @@ __global__ void UpdateDecoderMaskedSelfAttentionCacheIndirectionKernel(int32_t*
   const int batch_id = bb_id / beam_width;
   const int beam_id = bb_id % beam_width;
 
-  if (bb_id >= beam_width * batch_size || time_step >= current_length) {
+  if (bb_id >= beam_width * batch_size || time_step >= max_seq_length) {
```

Comment:
Can you please explain the changes here ? Why do we care about indirection indices for tokens after current_length ?

Reply:
it's just for debug purposes (to not see uninitialized values)

Reply:
That's true. Ideally, we would be memsetting the indirection buffer to all zeros before starting. That way, we can even avoid the first `if` where we assign the beam index to 0 for sequence indices < input sequence length. This is how FT does it. But, in any case, having un-initialized values in the indices >= current length should not matter as we shouldn't be using them anywhere.